src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -1139,9 +1154,22 @@ class CommonVisitor : public AST::BaseVisitor<Struct> {
         if (ASR::is_a<ASR::Function_t>(*s)) {
             ASR::Function_t *func = ASR::down_cast<ASR::Function_t>(s);
             if( n_kwargs > 0 && !is_generic_procedure ) {
-                args.reserve(al, n_pos_args + n_kwargs);
+                args.reserve(al, func->n_args);
                 visit_expr_list(pos_args, n_pos_args, kwargs, n_kwargs,
                                 args, rt_subs, func, loc);
+            } else if (args.size() < func->n_args) {
+                for (size_t def_arg = args.size(); def_arg < func->n_args; def_arg++){
+                    ASR::symbol_t* sym = ASR::down_cast<ASR::Var_t>(func->m_args[def_arg])->m_v;
+                    std::string variable_name = ASR::down_cast<ASR::Variable_t>(sym)->m_name;
+                    ASR::Variable_t* var = ASR::down_cast<ASR::Variable_t>(func->m_symtab->get_symbol(variable_name));
+                    if(var->m_symbolic_value == nullptr) {
+                        break;
+                    }
```

Comment:
Why do we `break` here? Is this a case where `m_symbolic_value` is expected/needed but is not available? If so, then I think we should throw a `SemanticError()` instead of the `break`.